phpunit/functional/SLMTest.php

```diff
@@ -1587,6 +1590,309 @@ public function testLaChange(): void
         }
     }
 
+
+    /**
+     * Escalation level changes when time passes for SLA TTR
+     */
+    public function testEscalationLevelChangesSlaTtr()
+    {
+        $this->login();
+        // create slm + sla ttr with 120 minutes
+        ['sla' => $sla] = $this->createSLA(data: [ 'number_time' => 120, 'definition_time' => 'minute',], sla_type: \SLM::TTR);
+
+        // add 2 escalation level to created SLA
+        $level_1 = $this->createItem(\SlaLevel::class, [
+            'name' => 'SLA level ' . time(),
+            'slas_id' => $sla->getID(),
+            'execution_time' => -60 * MINUTE_TIMESTAMP, // 60 minutes before TTR, 60 minutes elapsed
+            'is_active' => 1,
+            'is_recursive' => 1,
+            'match' => 'AND',
+        ]);
+        // criteria : ticket priority = 3
+        $this->createItem(\SlaLevelCriteria::class, ['criteria' => 'priority', 'condition' => 0, 'pattern' => '3', 'slalevels_id' => $level_1->getID()]);
+        // action : ticket priority -> 4
+        $this->createItem(\SlaLevelAction::class, ['action_type' => 'assign', 'field' => 'priority', 'value' => 4, 'slalevels_id' => $level_1->getID()]);
+
+        $level_2 = $this->createItem(\SlaLevel::class, [
+            'name' => 'SLA level ' . time(),
+            'slas_id' => $sla->getID(),
+            'execution_time' => -30 * MINUTE_TIMESTAMP, // 30 minutes before TTR, 90 minutes elapsed
+            'is_active' => 1,
+            'is_recursive' => 1,
+            'match' => 'AND',
+        ]);
+        // criteria : ticket priority = 4
+        $this->createItem(\SlaLevelCriteria::class, ['criteria' => 'priority', 'condition' => 0, 'pattern' => '4', 'slalevels_id' => $level_2->getID()]);
+        // action : ticket priority -> 5
+        $this->createItem(\SlaLevelAction::class, ['action_type' => 'assign', 'field' => 'priority', 'value' => 5, 'slalevels_id' => $level_2->getID()]);
+
+        // --- 10:00 - create ticket and affect slalevels_id_ttr
+        $this->setCurrentTime('10:00:00');
+        $this->runSlaCron(); // no changes will be triggered
```

Comment:
Maybe the cron be executed after the ticket creation, to validate that it does not alter the ticket.